src/mame/commodore/kim1.cpp

```diff
@@ -266,6 +302,13 @@ void kim1_state::sync_map(address_map &map)
 	map(0x0000, 0xffff).r(FUNC(kim1_state::sync_r));
 }
 
+// Called when serial data comes in from console.
+void kim1_state::tty_callback(uint8_t data)
+{
+	// Send data back to terminal to simulate the KIM-1 hardware
+	// echo.
+	m_rs232->write_txd(data);
+}
```

Comment:
You’re going to need to AND this with bit 0 of the port B output to avoid issues when they’re both changing at the same time.  Also, as this is a write line handler, the parameter type should conventionally be `int`.